### PR TITLE
add mbedtls for windows (since they build it now)

### PR DESCRIPTION
**Pull request by brendanf**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by jan-janssen** [comment]
@conda-forge-admin, please rerender

**Comment by brendanf** [comment]
This PR also manually pins to a version of mbedtls, which may be for the best since different versions are not binary compatible. (r-nanonext was failing because it was taking mbedtls=3.5 while nng was built against 3.3.)

**Comment by brendanf** [comment]
Pending review, this should be ready to go.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -22,10 +22,10 @@ requirements:
     - make
     - ninja
   host:
-    - mbedtls  # [not win]
+    - mbedtls =3.5
```
Comment (jan-janssen) [inline recipe/meta.yaml]: Is this required to be exactly `=3.5` or is it sufficient to use `>=3.5`.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -22,10 +22,10 @@ requirements:
...
   run:
     - ucrt  # [win]
-    - mbedtls  # [not win]
+    - mbedtls  =3.5
```
Comment (jan-janssen) [inline recipe/meta.yaml]: We could pin to the version which was used during building, using: 
```suggestion
    - {{ pin_compatible('mbedtls', max_pin='x.x.x') }}
```
https://conda-forge.org/docs/maintainer/pinning_deps.html

File: recipe/meta.yaml (hunk 1 above)
Comment (brendanf) [reply recipe/meta.yaml]: Now that I understand `pin_compatible`,  `>=3.5` should work.

File: recipe/meta.yaml (hunk 2 above)
Comment (brendanf) [reply recipe/meta.yaml]: Ah, I missed this in the pinning documentation.  That seems ideal.

**Review by jan-janssen — APPROVED** [review]
Looks good to me - @carlodri What do you think?

**Comment by carlodri** [comment]
It looks ok also for me, I'm merging, thanks @brendanf

**Pull request merged**